Sources/Datadog/Tracing/DDSpan.swift

```diff
@@ -84,33 +68,35 @@ internal class DDSpan: OTSpan {
     }
 
     func setOperationName(_ operationName: String) {
-        if warnIfFinished("setOperationName(_:)") {
-            return
+        queue.async {
+            if self.warnIfFinished("setOperationName(_:)") {
+                return
+            }
+            self.unsafeOperationName = operationName
         }
-        self.operationName = operationName
     }
 
     func setTag(key: String, value: Encodable) {
-        if warnIfFinished("setTag(key:value:)") {
-            return
-        }
-        ddTracer.queue.async {
+        queue.async {
+            if self.warnIfFinished("setTag(key:value:)") {
+                return
+            }
             self.unsafeTags[key] = value
         }
     }
 
     func setBaggageItem(key: String, value: String) {
-        if warnIfFinished("setBaggageItem(key:value:)") {
-            return
+        let isFinished = queue.sync { self.warnIfFinished("setBaggageItem(key:value:)") }
+        if !isFinished {
+            // Baggage items must accessed outside `tracer.queue` as it uses that queue for internal sync.
+            ddContext.baggageItems.set(key: key, value: value)
```

Comment:
💡 It is possible to synchronise `BaggageItems` access using `tracer.queue`, which would additionally clean up this code, leading to single `queue.async {}`. That would require small refactoring but I'm not sure if sharing single `tracer.queue` for all spans and baggage items is the best strategy. Alternatively, each `DDSpan` could work on its own `queue` and use it for baggage items synchronisation. The path to take highly depends on what do we optimise for: huge number of `DDSpans` (many queues) vs intensive use of `BaggageItems` (many accesses to single queue).

I stopped here, without going deeper in my refactoring, as the point of this PR is to clean up the interface of `DDSpan`. The solution in this PR already does less synchronization than before, so IMHO we can call it good enough and visit baggage items improvement on next occasion.